calculator.go

```diff
@@ -1,13 +1,92 @@
 // Package calculator provides a library for simple calculations in Go.
 package calculator
 
+import (
+	"errors"
+	"fmt"
+	"go/token"
+	"go/types"
+	"math"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
 // Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-return a + b
+func Add(a, b float64, extra ...float64) float64 {
+	var total float64 = a + b
+	for _, n := range extra {
+		total += n
+	}
+	return total
 }
 
 // Subtract takes two numbers and returns the result of subtracting the second
 // from the first.
-func Subtract(a, b float64) float64 {
-	return b - a
+func Subtract(a, b float64, extra []float64) float64 {
+	result := a - b
+	for _, n := range extra {
+		result -= n
+	}
+	return result
+}
+
+// Multiply takes two numbers and returns the result of multiplication them together.
+func Multiply(a, b float64, extra []float64) float64 {
+	result := a * b
+	for _, n := range extra {
+		result *= n
+	}
+	return result
+}
+
+// Divide takes two numbers and returns the result of divion and an error message
+func Divide(a, b float64, extra []float64) (float64, error) {
+	var divideByZeroError = "Cannot divide by zero"
+	if b == 0 {
+		return 0, errors.New(divideByZeroError)
```

Comment:
It's more efficient here to create an `error` value to use again, rather than a string variable:

```go
errDivideByZero := errors.New("cannot divide by zero")
...
return 0, errDivideByZero
```

(Sentinel errors, which is to say named error values, are conventionally named `ErrSomething`.)